packages/vat-data/src/vat-data-bindings.js

```diff
@@ -18,6 +21,12 @@ export {
   makeScalarWeakSetStore,
 };
 
+const { getEnvironmentOption } = makeEnvironmentCaptor(globalThis);
+const DETECT_PROVIDE_COLLISIONS =
+  getEnvironmentOption('DETECT_PROVIDE_COLLISIONS', 'false') === 'true';
```

Comment:
what ever sets this to true? if the goal is to help people become aware of a problem then they shouldn't need to know to enable the detection.

do we have any general ENV var for adopting development-time strictness? if not, time for one?

Reply:
@samsiegart , what do each of the following lines do?

https://github.com/Agoric/agoric-sdk/blob/995beed49c308dc80b92b29a077192b037ca053f/packages/wallet/ui/src/lockdown.js#L4

https://github.com/Agoric/dapp-card-store/blob/67d2c946130ae1510507f93c18768c9497c5e5eb/ui/src/install-ses-lockdown.js#L9

For the latter, I understand the notation, but not the comment above the line. But where does this setting of `NODE_ENV` come from?

For both, since it is checking whether it is set to `'production'`, does that mean that in the absence of this unexplained flag we default to unsafe settings? In general, the default way should be the safe way. An explicit override should be needed to sacrifice safety for better diagnostics.

Reply:
`NODE_ENV` is provided by `react-scripts`. 

We can do `const consoleTaming = '%NODE_ENV%' === 'development' ? 'unsafe' : 'safe';`

In development mode, react devtools tries to extend `console` to add debugging features iirc

Reply:
> We can do `const consoleTaming = '%NODE_ENV%' === 'development' ? 'unsafe' : 'safe';`

Yes, please do. That way, if it is absent we'll default to safe. Implicit defaults should always be safe.

Reply:
@samsiegart , see https://github.com/Agoric/agoric-sdk/pull/6079